src/compactor_executor.rs

```diff
@@ -96,7 +96,7 @@ impl TokioCompactionExecutorInner {
         let l0_merge_iter = MergeIterator::new(l0_iters).await?;
         let mut sr_iters = VecDeque::new();
         for sr in compaction.sorted_runs.iter() {
-            let iter = SortedRunIterator::new_spawn(sr, self.table_store.clone(), 4, 256).await?;
+            let iter = SortedRunIterator::new_spawn(sr, self.table_store.clone(), 16, 256).await?;
```

Comment:
Is this something that should be configurable? I'm guessing you updated it based on the benchmark tests. If you don't think it needs to go much higher, we can probably leave it.

Aside: did you play with bumping blocks_to_fetch vs. max_fetch_tasks? Wonder how these two interplay. There's probably some happy medium...

Reply:
yeah, I bumped it based on my testing. I hit this problem where the db was compacting a smaller SST with a larger SST, and was bottlenecked on reading the larger SST - so I bumped this value to make sure there were always buffers ready to be compacted (at the cost of some extra memory taken up by the buffer). I'm not opposed to making this configurable - I think we should allow the user to configure the total buffer size allowed per SST or in total. That said, this seems to work well enough based on the testing I've done so far.

> Aside: did you play with bumping blocks_to_fetch vs. max_fetch_tasks? Wonder how these two interplay. There's probably some happy medium...

I did - I think around 1MB you start to get more benefit from fetching in parallel rather than streaming more data from one request - hence the 256 value here.

Reply:
> That said, this seems to work well enough based on the testing I've done so far.

💯 I'd prefer to expose the least  amount as possible for as long as possible until people actually can justify needing to configure it.